src/status_im2/contexts/chat/photo_selector/view.cljs

```diff
@@ -41,44 +40,40 @@
            (i18n/label :t/confirm-selection)]])))])
 
 (defn clear-button
-  []
-  (when (seq @selected)
-    [rn/touchable-opacity
-     {:on-press            #(reset! selected [])
-      :style               (style/clear-container)
-      :accessibility-label :clear}
-     [quo/text {:weight :medium} (i18n/label :t/clear)]]))
-
-(defn remove-selected
-  [coll item]
-  (vec (remove #(= % item) coll)))
+  [chat-id]
+  (let [selected (rf/sub [:chats/selected-photos])]
+    (when (seq selected)
+      [rn/touchable-opacity
+       {:on-press            #(rf/dispatch [:chat.ui/clear-sending-images chat-id])
+        :style               (style/clear-container)
+        :accessibility-label :clear}
+       [quo/text {:weight :medium} (i18n/label :t/clear)]])))
 
 (defn image
-  [item index _ {:keys [window-width]}]
-  [rn/touchable-opacity
-   {:active-opacity      1
-    :on-press            (fn []
-                           (if (some #{item} @selected)
-                             (swap! selected remove-selected item)
-                             (swap! selected conj item)))
-    :accessibility-label (str "image-" index)}
-   [rn/image
-    {:source {:uri item}
-     :style  (style/image window-width index)}]
-   (when (some #{item} @selected)
-     [rn/view {:style (style/overlay window-width)}])
-   (when (some #{item} @selected)
-     [info-count/info-count
-      {:style               style/image-count
-       :accessibility-label (str "count-" index)}
-      (inc (utils/first-index #(= item %) @selected))])])
+  [item index _ {:keys [window-width chat-id]}]
+  (let [selected (rf/sub [:chats/selected-photos])]
+    [rn/touchable-opacity
+     {:active-opacity      1
+      :on-press            (fn []
+                             (if (some #{item} selected)
+                               (rf/dispatch [:chat.ui/image-unselected item])
+                               (rf/dispatch [:chat.ui/image-selected chat-id item])))
+      :accessibility-label (str "image-" index)}
```

Comment:
Do we need to define these dynamic accessibility labels from the point of view of Appium tests?

Actually the question goes first to @churik: Can the QA team reliably select elements on screen based on an index? Example: if there are two instances of a component with accessibility label `X`, can we write a selector in Appium to get the second one?

For this particular example in the PR, the accessibility label could be just `image`, and then in Appium you would be able to select the image based on an index. Selecting by index is supported in most major e2e test tools that I know of (Selenium based or not), but I don't have experience with Appium.

If what I'm suggesting is possible and reliable, then we could eliminate most dynamic accessibility labels in the code.

Reply:
sure we can, we just need to have an anchor.

Reply:
Cool @churik, thanks for the info.

@ibrkhalil, this means we don't need to create accessibility labels like: `image-1`, `image-2`, and so on. All we need is to define the accessibility level `image` or any other unique name on the screen, and the QA team will be able to select based on an index. Just to confirm, is this correct @churik?

So, something along these lines for dev guidelines:

```clojure
;; bad
:accessibility-label (str "image-" index)

;; good
:accessibility-label :image
```

So I suggest we start following this practice from now on. Any objections @flexsurfer?

Reply:
thanks Icaro, no objections